Support DensityInterface v0.4

oschulz opened the pull request:
This is non-breaking and keeps support for DensityInterface v0.3.2.

devmotion reviewed (commented):
Thanks for the PR! Is there a good reason to keep support for < 0.4? I would assume it is easier to just switch to 0.4.

Can you also update the version number of Distributions?

oschulz commented:
> I would assume it is easier to just switch to 0.4.

I guess so, not many packages depend on DensityInterface so far.

Ok, dropped DensityInterface v0.3 and updated Distributions version number.

devmotion commented on src/density_interface.jl:
```diff
@@ -1,4 +1,4 @@
-@inline DensityInterface.hasdensity(::Distribution) = true
+@inline DensityInterface.DensityKind(::Distribution) = HasDensity()
```
I think it's not exported, is it?
```suggestion
@inline DensityInterface.DensityKind(::Distribution) = DensityInterface.HasDensity()
```

oschulz replied:
Hm, it actually *is* exported:

https://github.com/JuliaMath/DensityInterface.jl/blob/e71eb2ded6f3320d66a6d7efc615e3e60dddb58f/src/interface.jl#L69

oschulz replied:
Ah, but we "import DensityInterface" :-)

devmotion reviewed (approved):
Thank you!

oschulz commented:
Thanks!
